app/src/main/java/com/fbada/projectone/Car.kt

```diff
@@ -0,0 +1,349 @@
+        println(specs)
+    }
+    /**
+     * @property price Base price plus Sprinter Rating Price and Sprinter Max Speed Price
+     */
+    override var price: Double = super.price + sprinterRatingPrice() + sprinterMaxSpeedPrice()
+    /**
+     * Calculates a price based on customer rating
+     */
+    fun sprinterRatingPrice(): Double {
+        val extras: Double = when (rating) {
+            in 0..10 -> 2.00 + rating
+            in 11..30 -> 6.00 + rating
+            in 31..55 -> 10.00 + rating
+            in 56..77 -> 15.00 + rating
+            else -> 22.00 + rating
+        }
+        return extras
+    }
+    /**
+     * Calculates a price based on vehicle max speed.
+     */
+    fun sprinterMaxSpeedPrice(): Double {
+        val extras: Double = when (maxSpeed) {
+            in 0..60 -> 10.00
+            in 61..80 -> 20.00
+            in 81..100 -> 30.00
+            else -> 50.00
+        }
+        return extras
+    }
+}
+/**
+ * Subclass Traviliner Model type car.  Extends Mercedes class.
+ *  @property rating Customer rating score.
+ *  @property maxSpeed Maximum speed of vehicle.
+ */
+class Traveliner(private val rating: Int, val maxSpeed: Int) : Mercedes(6, 1) {
+    /**
+     * @param rating Customer rating score.
+     * @param maxSpeed Maximum speed of vehicle.
+     * @param gearBox Gearbox type.
+     */
+    constructor(rating: Int, maxSpeed: Int, gearBox: String) : this(rating, maxSpeed) {
+        val specs =
+            "This Traveliner car has the following specs:\n Customer Rating:$rating\n Max Speed:$maxSpeed \n Number of Doors:$doors \n Previous Owners $owners"
+        println(specs)
+    }
+    /**
+     * @property price Base price plus max speed price calculation.
+     */
+    override var price: Double = super.price + travelinerMaxSpeedPrice()
+    /**
+     * Calculates a price based on max speed.
+     */
+    fun travelinerMaxSpeedPrice(): Double {
+        val extras: Double = when (maxSpeed) {
+            in 0..60 -> 10.00
+            in 61..80 -> 20.00
+            in 81..100 -> 30.00
+            else -> 50.00
+        }
+        return extras
+    }
+
+}
+/**
+ * Subclass Cougar Model type car.  Extends Ford class.
+ * @property interior Interior material type.
+ * @property materialPrice Price of interior material.
+ * @property spareTyre Spare Tyre available.
+ * @property radioType Type of radio inside car.
+ */
+class Cougar(private val interior: String, private var materialPrice: Double, val spareTyre: Boolean, val radioType: String) :
+    Ford() {
+    /**
+     * @property price Base price plus interior calculation
+     */
+    override var price: Double = super.price + interiorPremium()
+    /**
+     * Calculates a price based on interior.  If the interior is Leather or Luxuary the price is multiplied by 2.5.
+     */
+    fun interiorPremium(): Double {
```

Comment:
Same as above. Feel free to resolve if you plan on exposing those kind of functions.
But there should be a strong explanation on why we would need to do that.